tensorflow/python/ops/metrics_impl.py

```diff
@@ -2130,6 +2130,115 @@ def compute_recall(true_p, false_n, name):
+  `update_op` that updates these variables and returns the `specificity`.
+  `update_op` weights each prediction by the corresponding value in `weights`.
+
+  If `weights` is `None`, weights default to 1. Use weights of 0 to mask values.
+
+  For additional information about specificity and sensitivity, see the
+  following: https://en.wikipedia.org/wiki/Sensitivity_and_specificity
+
+  Args:
+    labels: The ground truth values, a `Tensor` whose dimensions must match
+      `predictions`. Will be cast to `bool`.
+    predictions: The predicted values, a `Tensor` of arbitrary dimensions. Will
+      be cast to `bool`.
+    weights: Optional `Tensor` whose rank is either 0, or the same rank as
+      `labels`, and must be broadcastable to `labels` (i.e., all dimensions must
+      be either `1`, or the same as the corresponding `labels` dimension).
+    metrics_collections: An optional list of collections that `specificity`
+      should be added to.
+    updates_collections: An optional list of collections that `update_op` should
+      be added to.
+    name: An optional variable_scope name.
+
+  Returns:
+    specificity: Scalar float `Tensor` with the value of `true_negatives`
+      divided by the sum of `true_negatives` and `false_positives`.
+    update_op: `Operation` that increments `true_negatives` and
+      `false_positives` variables appropriately and whose value matches
+      `specificity`.
+
+  Raises:
+    ValueError: If `predictions` and `labels` have mismatched shapes, or if
+      `weights` is not `None` and its shape doesn't match `predictions`, or if
+      either `metrics_collections` or `updates_collections` are not a list or
+      tuple.
+    RuntimeError: If eager execution is enabled.
+  """
+  if context.in_eager_mode():
+    raise RuntimeError('tf.metrics.recall is not supported is not '
+                       'supported when eager execution is enabled.')
+
+  with variable_scope.variable_scope(name, 'specificity',
+                                     (predictions, labels, weights)):
+    predictions, labels, weights = _remove_squeezable_dimensions(
+        predictions=math_ops.cast(predictions, dtype=dtypes.bool),
+        labels=math_ops.cast(labels, dtype=dtypes.bool),
+        weights=weights)
+
+    true_n, true_negatives_update_op = true_negatives(
+        labels,
+        predictions,
+        weights,
+        metrics_collections=None,
+        updates_collections=None,
+        name=None)
+    false_p, false_positives_update_op = false_positives(
+        labels,
+        predictions,
+        weights,
+        metrics_collections=None,
+        updates_collections=None,
+        name=None)
+
+    def compute_specificity(true_n, false_p, name):
+      return array_ops.where(
+          math_ops.greater(true_n + false_p, 0),
+          math_ops.div(true_n, true_n + false_p), 0, name)
+
+    specificity = compute_specificity(true_n, false_p, 'value')
```

Comment:
Note that this should be updated to reflect the changes to this file to support distributed.